utils/auction.ts

```diff
@@ -65,6 +65,19 @@ export const getUniqueTraders = (bids: Array<Bid>) => {
   return Object.keys(uniqueApprovalMap)
 }
 
+export const checkNonces = async (bids: Array<Bid>) => {
+  const nonceCheckPromises = bids.map(async bid => {
+    const isNonceUsed = await crabV2Contract.nonces(bid.bidder, bid.order.nonce)
+    return { bidder: bid.bidder, nonce: bid.order.nonce, isNonceUsed }
+  })
+
+  const nonceResults = await Promise.all(nonceCheckPromises)
+  return nonceResults.reduce((acc, { bidder, nonce, isNonceUsed }) => {
+    acc[`${bidder}-${nonce}`] = isNonceUsed
+    return acc
+  }, {} as { [key: string]: boolean })
```

Comment:
Question from my javascript ignorance: do we need to give {} as { [key : string]: boolean} for the initial value of the accumulator or could we just give empty array for the reduce?

Reply:
here we are dealing with objects, so need to give empty object ie {}. if it'd have been an array, then empty array would have worked

Reply:
that makes sense but would it work if we just gave it {} like 

```
  return nonceResults.reduce((acc, { bidder, nonce, isNonceUsed }) => {
    acc[`${bidder}-${nonce}`] = isNonceUsed
    return acc
  }, {} as { [key: string]: boolean })
```